src/Employer/Employer.Web/Controllers/Part1/TrainingProviderController.cs

```diff
@@ -28,19 +29,27 @@ public TrainingProviderController(TrainingProviderOrchestrator orchestrator, IRe
         [HttpGet("select-training-provider", Name = RouteNames.TrainingProvider_Select_Get)]
         public async Task<IActionResult> SelectTrainingProvider(VacancyRouteModel vrm, [FromQuery] string wizard = "true", [FromQuery] string clear = "", [FromQuery] long? ukprn = null)
         {
-            var vm = await _orchestrator.GetSelectTrainingProviderViewModelAsync(vrm, ukprn);
-            vm.PageInfo.SetWizard(wizard);
-
-            if (string.IsNullOrWhiteSpace(clear) == false)
-            {
-                vm.Ukprn = string.Empty;
-                vm.TrainingProviderSearch = string.Empty;
-                vm.IsTrainingProviderSelected = true;
-            }
-            
+           var vm = await _orchestrator.GetSelectTrainingProviderViewModelAsync(vrm, ukprn);
+           vm.PageInfo.SetWizard(wizard);
+
+           if (string.IsNullOrWhiteSpace(clear) == false)
+           {
+               vm.Ukprn = string.Empty;
+               vm.TrainingProviderSearch = string.Empty;
+               vm.IsTrainingProviderSelected = true;
+           }
+           vm.ReferredFromMAHome_FromSavedFavourites = ShowReferredFromMABackLink();
```

Comment:
simplify this terms of showing/hiding a back link.  The models and views shouldn't care about MA and favourites etc.